framework-plugins/lisk-framework-monitor-plugin/src/monitor_plugin.ts

```diff
@@ -223,6 +233,17 @@ export class MonitorPlugin extends BasePlugin {
 		}
 	}
 
+	private _handleFork(block: string) {
+		const { header } = codec.decode<RawBlock>(this.schemas.block, Buffer.from(block, 'hex'));
+		const decodedHeader = codec.decode<RawBlockHeader>(this.schemas.blockHeader, header);
+		const blockId = hash(header).toString('hex');
+		this._state.forks.forkEventCount += 1;
+		this._state.forks.blockHeaders[blockId] = {
+			blockHeader: decodedHeader,
+			timeReceived: Date.now(),
+		};
```

Comment:
`this._state.forks.blockHeaders[blockId]` maybe check if the blockId exists or not and only update the time if it exists. In that case we don't have to do decoding